Sources/VitalHealthKit/HealthKit/VitalHealthKitClient.swift

```diff
@@ -283,7 +280,7 @@ extension VitalHealthKitClient {
 
   @available(iOS 15.0, *)
   private func bundledBackgroundObservers(
-    for typesBundle: Set<[HKSampleType]>
+    for typesBundle: Set<Set<HKSampleType>>
```

Comment:
Why would we want a set of sets?

Reply:
For each VitalResource, there shall be a unique set of sample types waiting to be observed. So `Set<Set<T>>` is arguably the correct semantic.

Though I want to note that this is temporary. The ultimate goal is to refactor this to `[VitalResource: Set<HKSampleType>]` so that we can eliminate the need of the brittle HKSampleType -> VitalResource mapping altogether (the site of some of the crashes).

Reply:
See #56 for the final stage.

Reply:
Why is there a crash?